addon/-private/system/many-array.js

```diff
@@ -136,7 +136,12 @@ export default Ember.Object.extend(Ember.MutableArray, Ember.Evented, {
   objectAt(index) {
     let object = this.currentState[index];
     //Ember observers such as 'firstObject', 'lastObject' might do out of bounds accesses
-    if (object === undefined) { return; }
+    if (object === undefined) {
```

Comment:
this needs to bounds check that `index` is not `length-1` or `0` because of the greediness of `firstObject` and `lastObject`.